endpoints/internal-check-file-acl.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Automattic\VIP\Files\Acl;
+
+require_once __DIR__ . '/../files/acl.php';
+
+$file_request_uri = $_SERVER['HTTP_X_ORIGINAL_URI'] ?? null;
+
+$validated_file_path = pre_wp_validate_request( $file_request_uri );
+if ( is_array( $validated_file_path ) ) {
+	pre_wp_send_error_headers( $validated_file_path );
+
+	exit;
+}
+
+// TODO: make sure HTTP HOST and path have subdir set for multisite
+// TODO: handle resized files (e.g. file-200x200.jpg)
+
+// Bootstap WordPress
+require __DIR__ . '/../../../wp-load.php';
```

Comment:
I think you can define `SHORTINIT` as `true` before here, and still have the `apply_filters()` work, though that would then limit what functions are available to be used in the filter callbacks.